proto/internal/temporal/server/api/historyservice/v1/request_response.proto

```diff
@@ -427,6 +427,16 @@ message ReplicateEventsV2Request {
 message ReplicateEventsV2Response {
 }
 
+message ReplicateHistoryEventsRequest {
+    string namespace_id = 1;
+    temporal.api.common.v1.WorkflowExecution workflow_execution = 2;
+    repeated temporal.server.api.history.v1.VersionHistoryItem version_history_items = 3;
+    repeated temporal.api.history.v1.HistoryEvent events = 4;
+    // New run events does not need version history since there is no prior events.
+    repeated temporal.api.history.v1.HistoryEvent new_run_events = 5;
+    temporal.server.api.workflow.v1.BaseExecutionInfo base_execution_info = 6;
+}
+
```

Comment:
i am ok using proto definition to define a request struct, just pointing out ^ is not required, since this new API is not a RPC API

Reply:
Yeah, that's a good point. I changed the API to accept raw data instead of a struct.